### ISSUE TITLE
Tweak the dataset documentation function of the `export_data()` function to account for the new database structure

### ISSUE THREAD

**Issue by BBieri** [globalgov/qData#92]
The current `export_data()` function does not account for the new dataset structure (as a list of dataset rather than a set of independent datasets) in its dataset documentation creation. 

### Example : 
``` r
% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/states.R
\docType{data}
\name{states}
\alias{states}
\title{Data on the entry and exit of sovereign states to the interstate system containing
the following datasets:}
\format{
This dataset contains 243 observations and 5 variables:
\describe{
\item{COW_Nr}{Numerical COW identifier}
\item{ID}{# letter abbreviation of State name}
\item{Beg}{Beginning	date	of	state	tenure}
\item{End}{End date of state tenure}
\item{Label}{Primary COW State name}
}
}
\source{
\url{https://correlatesofwar.org/data-sets/state-system-membership}
}
\usage{
data(states)
}
\description{
Data on the entry and exit of sovereign states to the interstate system containing
the following datasets:
}
\keyword{datasets}

```

**Comment by jhollway** [comment]
Does the documentation all need to be together in one script to be roxygen compliant?

**Comment by BBieri** [comment]
Sorry for not seeing your comment earlier. Basically, I am trying to figure out the best way to document our list of datasets. I am currently trying to implement both an individual level dataset description as well as a database description which is essentially a list of individual datasets (uses `\enumerate` and `\item`). 

Taking inspiration from [this post](https://stackoverflow.com/questions/38095578/documenting-a-list-of-data-frames-with-roxygen2).

**Comment by BBieri** [comment]
### Update:

There are essentially two alternatives to implement `{roxygen}` documentation in our project.

1. Create a unique documentation script per database that is very detailed, e.g. contains both database and dataset level information. Think something along these lines:
```` r
#'states
#' The states database contains 3 datasets: COW, GW, ISD
#' @format A list of 3 dataframes.
#' \describe{
#' \item{COW dataset containing 243 observations and 5 variables}{Describe your dataset here}
#' Detailed documentation of all variables of COW.
#' \item{GW dataset containing 216 observations and 5 variables}{Describe your dataset here}
#' Detailed documentation of all variables of GW.
#' \item{ISD dataset containing 362 observations and 7 variables}{Describe your dataset here}
#' Detailed documentation of all variables of ISD.
#' }
"states"

````
2. Export individual datasets as well and create documentation for them separately. We would keep the existing template for the datasets and add a summary documentation for the database. Think something along the lines of for databases: 
```` r
#'states
#' The states database contains 3 datasets: COW, GW, ISD
#' @format A list of 3 dataframes.
#' \describe{
#' \item{COW dataset containing 243 observations and 5 variables}{Describe your dataset here}
#' \item{GW dataset containing 216 observations and 5 variables}{Describe your dataset here}
#' \item{ISD dataset containing 362 observations and 7 variables}{Describe your dataset here}
#' }
"states"
````


__To recap': __
- Either we create a single very detailed documentation file at the database level and only keep databases in the data folder
- Or we keep both databases _and_ datasets in the data folder and create both a high-level dataset documentation and a more detailed dataset documentation

**Comment by jhollway** [comment]
What are the trade-offs? If one can call individual help pages, it might be more efficient and user-friendly to have option two.

**Comment by BBieri** [comment]
The only tradeoff of option 2 would be that the data folder of each data package would need to contain both individual datasets as well as lists of datasets.

**Comment by jhollway** [comment]
Ah, so this there is no way to have the distributed documentation without duplicating the data?

**Comment by henriquesposito** [comment]
I also like option 2 better. Though there are some redundancies (i.e. some information is repeated), option 2 is a bit more transparent and more information is better than less here, in my opinion. If I am not mistaken the dataset level documentation it exports' only the database object, not the dataset. This, in turn, avoids that we create objects for datasets and for database, even when creating documentation pages for both.

**Comment by BBieri** [comment]
@jhollway The issue here is that we can't documents objects that are not present in the `/data` folder. Hence, we have the choice of either adding individual datasets to the  `/data` folder and write one documentation file for each object (datasets and databases) or only keep the database object in the `/data`  folder and make one big documentation file for the database containing a detailed description of all datasets. 

@henriquesposito and I favor the latter since this mirrors the package's philosophy more and with the inbuilt search function of the R-Studio help panel, users can easily search for the relevant information by typing "COW dataset" for example.

**Comment by jhollway** [comment]
But there could be multiple versions of the same dataset in the package, e.g. `agreements$IEADB` and `memberships$IEADB` in `{qEnviron}`.

**Comment by BBieri** [comment]
In this instance, there would be two data objects in the `/data` folder: `agreements.rda` and `memberships.rda`. Hence both objects (aka databases) would be documented by a separate documentation file which would contain a section IEADB Membership Dataset for the membership database and IEADB Agreements dataset section for the Agreements database. Both of these sections would be different.

**Comment by BBieri** [comment]
@jhollway @henriquesposito  @EstherPeev Pushed changes that creates/updates a metadata object for the created/updated database object when `eport_data()` runs. I would love feedback on the following: 
- Overall idea of having this companion list of lists with various metadata. Do you find it useful?
- Elements that should be in this metadata object. I.e. what do you want to know as a user?

In fine, my idea was to source as much information for the documentation from this object. We talked at the end of last week about doing a custom documentation function since `{roxygen}` does not work if the objects are not in the data folder. This would thus be the starting point of this endeavor. Furthermore, I think it could also be interesting to look into it further for other data comparison functions.
